site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -588,4 +588,12 @@ private function sendSocketMessage(array $msg_array): void {
             $this->core->addNoticeMessage("WebSocket Server is down, page won't load dynamically.");
         }
     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/office_hours_queue/preview", methods={"POST"})
+     * @AccessControl(role="LIMITED_ACCESS_GRADER")
+     */
+    public function showMarkdownPreview() {
+        $this->core->getOutput()->renderOutput('OfficeHoursQueue', 'previewAnnouncement', $_POST['enablePreview'], $_POST['content']);
```

Comment:
Two things:
1. The `$_POST['enablePreview']` should be snake_case (`$_POST['enable_preview']`) to follow convention around the codebase.
2. We want our controllers to return a value, and ideally not be plugging directly into the render class. This makes it easier to test, and allowing our rendering to happen in one place (`public/index.php`).

Putting this together:
```suggestion
        return new WebResponse(OfficeHoursQueue::class, 'previewAnnouncement', $_POST['enable_preview'], $_POST['content']);
```